### PR TITLE
feat: add svdb workflow

### PR DESCRIPTION
**Pull request by khurrammaqbool**
### This PR:

Added: SVDB to the workflow

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by khurrammaqbool** [comment]
Screenshot 2022-03-04 at 15 04 23

Screenshot 2022-03-04 at 15 08 45

Screenshot 2022-03-04 at 15 09 09

Screenshot 2022-03-04 at 15 06 01

Screenshot 2022-03-04 at 15 07 13

Screenshot 2022-03-04 at 15 07 53

**Review by ashwini06 — CHANGES_REQUESTED** [review]
Good job 👍🏼  

Just some general comments about the results from manta and delly. If we are going to only store and upload merged svdb results, do we need the annotation of manta/delly using vep and also existing rules for `bcftools_filter_delly` and `bcftools_filter_manta`?

**Comment by khurrammaqbool** [comment]
> Good job 👍🏼
> 
> Just some general comments about the results from manta and delly. If we are going to only store and upload merged svdb results, do we need the annotation of manta/delly using vep and also existing rules for `bcftools_filter_delly` and `bcftools_filter_manta`?

I agree and already planned to handle it in next PR as the current only adds svdb.

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule [hunk 1]
```diff
@@ -103,5 +103,30 @@ bcftools view  --threads {threads} -f PASS -O z -o {output.vcf} {input.bcf};
...
+    benchmark:
+        Path(benchmark_dir, 'svdb_merge_tumor_only_' + config["analysis"]["case_id"] + ".tsv")
+    singularity:
+        Path(singularity_image, config["bioinfo_tools"].get("svdb") + ".sif").as_posix()
+    params:
+        tumor = get_sample_type(config["samples"], "tumor"),
+        case_name = config["analysis"]["case_id"],
+        var_caller = "svdb"
+    threads:
+        get_threads(cluster_config, "svdb_merge_tumor_only")
```
[the remark this answers is not in the document]
Comment (khurrammaqbool) [reply BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule:122]: I am waiting for the benchmarks.

File: BALSAMIC/snakemake_rules/annotation/varcaller_sv_wgs_filter_tumor_only.rule [hunk 2]
```diff
@@ -51,3 +51,27 @@ rule bcftools_filter_delly:
 bcftools view --threads {threads} -f .,PASS -o {output.vcf_sv_pass} -O z {input.vcf};
 tabix -p vcf -f {output.vcf_sv_pass};
     """
+
+rule bcftools_filter_svdb:
+  input:
+    vcf = vep_dir + "SV.somatic.{case_name}.svdb.all.vcf.gz",
+  output:
+    vcf_sv_pass = vep_dir + "SV.somatic.{case_name}.svdb.all.filtered.pass.vcf.gz",
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/annotation/varcaller_sv_wgs_filter_tumor_only.rule:59]:
```suggestion
    vcf_sv_pass_svdb = vep_dir + "SV.somatic.{case_name}.svdb.all.filtered.pass.vcf.gz",
```

File: BALSAMIC/snakemake_rules/annotation/varcaller_sv_wgs_filter_tumor_normal.rule [hunk 3]
```diff
@@ -75,3 +75,27 @@ rule bcftools_filter_ascat:
 bcftools view --threads {threads} -f .,PASS -o {output.vcf_sv_pass} -O z {input.vcf};
 tabix -p vcf -f {output.vcf_sv_pass};
     """
+
+rule bcftools_filter_svdb:
+  input:
+    vcf = vep_dir + "SV.somatic.{case_name}.svdb.all.vcf.gz",
+  output:
+    vcf_sv_pass = vep_dir + "SV.somatic.{case_name}.svdb.all.filtered.pass.vcf.gz",
```
Comment (ivadym) [inline BALSAMIC/snakemake_rules/annotation/varcaller_sv_wgs_filter_tumor_normal.rule:83]:
```suggestion
    vcf_sv_pass_svdb = vep_dir + "SV.somatic.{case_name}.svdb.all.filtered.pass.vcf.gz",
```

**Review by ivadym — CHANGES_REQUESTED** [review]
The tags that BALSAMIC produces should be uniquely identified by Hermes. Some ways to do it: change the output variable name (which is included as a tag), use `svdb` as a wildcard or add an additional tag for the `housekeeper_id` variable.

File: BALSAMIC/snakemake_rules/annotation/varcaller_sv_wgs_filter_tumor_normal.rule [hunk 4]
```diff
@@ -75,3 +75,27 @@ rule bcftools_filter_ascat:
...
+  benchmark:
+    benchmark_dir + "bcftools_filter_svdb_SV.somatic.{case_name}.svdb.vep.tsv"
+  singularity:
+    Path(singularity_image,config["bioinfo_tools"].get("bcftools") + ".sif").as_posix()
+  params:
+    case_name = "{case_name}",
+    housekeeper_id = {"id": config["analysis"]["case_id"], "tags": "clinical"},
+    var_caller = "svdb"
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/annotation/varcaller_sv_wgs_filter_tumor_normal.rule:91]:
```suggestion

```

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 5]
```diff
@@ -206,3 +206,33 @@ rule ascat_tumor_normal_merge_output:
...
+        namemap = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".svdb.sample_name_map",
+    benchmark:
+        Path(benchmark_dir, 'svdb_merge_tumor_normal_' + config["analysis"]["case_id"] + ".tsv")
+    singularity:
+        Path(singularity_image, config["bioinfo_tools"].get("svdb") + ".sif").as_posix()
+    params:
+        tumor = get_sample_type(config["samples"], "tumor"),
+        normal = get_sample_type(config["samples"], "normal"),
+        case_name = config["analysis"]["case_id"],
+        var_caller = "svdb"
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:225]:
```suggestion

```

**Review by ashwini06 — CHANGES_REQUESTED** [review]
small fixes to adress

**Review by ashwini06 — APPROVED** [review]

**Review by ivadym — APPROVED** [review]
👍

**Pull request merged**